.github/workflows/test_nix.yml

```
@@ -0,0 +1,41 @@
+name: Test Nix
+permissions:
+  contents: read
+
+on: [push]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
+    # container:
+    #   image: nixos/nix
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - name: Cache Nix
+        id: nix-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            /nix/store
+            ~/.nix-profile/.nix-profile
+          key: nix-rust-${{ hashFiles('**/flake.lock') }}
+          restore-keys: nix-rust-
+
+      - name: Install Nix
+        if: ${{ steps.nix-cache.outputs.cache-hit != 'true' }}
+        run: curl -L https://nixos.org/nix/install | sh -s -- --daemon
+
+      - name: Set Nix Path
+        run: |
+          # Set nix paths gotten from
+          # https://github.com/cachix/install-nix-action/blob/5cfd5166ea7680b784fd8cf556c6d07a9ecce260/install-nix.sh#L85
+          echo "/nix/var/nix/profiles/default/bin" >> "$GITHUB_PATH"
+          # new path for nix 2.14
+          echo "$HOME/.nix-profile/bin" >> "$GITHUB_PATH"
+          if [[ -n "${INPUT_NIX_PATH:-}" ]]; then
+            echo "NIX_PATH=${INPUT_NIX_PATH}" >> "$GITHUB_ENV"
+          fi
+
```

As we discussed, before we merge anything let's replace these with:
```suggestion
      - name: Install Nix
        # https://github.com/DeterminateSystems/nix-installer-action/releases/tag/v4
        uses: DeterminateSystems/nix-installer-action@65d7c888b2778e8cf30a07a88422ccb23499bfb8
        # https://github.com/DeterminateSystems/magic-nix-cache-action/releases/tag/v2
      - name: Cache Nix Store Incrementally
        uses: DeterminateSystems/magic-nix-cache-action@8a218f9e264e9c3803c9a1ee1c30d8e4ab55be63
```